### ISSUE TITLE
Bug: [promise-function-async] Support for await calls in template literals

### ISSUE THREAD

**Issue by Code-Hex** [typescript-eslint/typescript-eslint#8058]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.2&fileType=.ts&code=MYewdgzgLgBAZmGBeGBDCBPMwYAoCUyAfDAAoBOIAtgJYQCmAdOfRCADYBu9uA5ABL127EL3wBuAFCT0WHHACu2KDXDwlwFeABK9KAvKQK1Oj0IBvSTBgt9hmAAMAJOYQEAvg-EwA9D5j0AB4ADvSa9AAmMC6oAO6oNLBu%2BO6SqUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6AewFtLlFaAM1hMy%2BSpya0AhskKj0URNC7RI4MAF8QmoA&tsconfig=&tokens=false

### Repro Code

```TypeScript
const fn = async () => Promise.resolve('Hello');

async function functionReturnsPromise() {
  return `${fn()}`; // expected ${await fn()}
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/promise-function-async": "error"
  }
}
```

### Expected Result

`${fn()}`; // expected ${await fn()}

### Actual Result

No error

### Additional Info

https://github.com/honojs/hono/issues/1812

**Comment by bradzacher** [comment]
I think you might have your wires crossed - the `promise-function-async` rule exists to enforce that functions that return a promise are marked with `async`.
It has nothing to do with checking for `await`.

Was there a different rule you were thinking of?

**Comment by Code-Hex** [comment]
@bradzacher  I see, that makes sense. I had misunderstood the behavior of this plugin. Indeed, the corrections it makes simply involve adding the `async` keyword.

So, I should take care of `await` on my judgment.

Thank you for clarifying.

**Comment by Josh-Cena** [comment]
You may be interested in `no-misused-promises`, `restrict-template-expressions`, and `no-base-to-string` :)
